app/lib/features/space/widgets/top_nav.dart

```diff
@@ -46,13 +48,12 @@ class _TopNavBarState extends ConsumerState<TopNavBar>
 
   @override
   Widget build(BuildContext context) {
+    ref.watch(selectedTabIdxProvider(widget.spaceId));
     final tabState = ref.watch(recentWatchScreenTabStateTestProvider(context));
-    final selectedIndex = ref.watch(selectedTabIdxProvider(widget.spaceId));
     return tabState.when(
       data: (tabState) {
         final tabController = tabState.ctrl;
         final tabs = tabState.tabs;
```

Comment:
it works without the animate? Are we sure this works when the tab-route change is not coming from the tab-bar but from the outside routing? e.g. when the user clicks on an overview widget that leads to a sub-tab?

Reply:
To reduce flickering effect without using any animation, I have did some of the changes in `top_nav.dart` as well.

1. Get rid of initialising `TabController` with `initialIndex:0` and opening specific tab with selectedIndex in build method with `tabController.animateTo(selectedIndex);`
2. Use below code in `initState` to handle tab selection.
```
@override
  void initState() {
    super.initState();
    recentWatchScreenTabStateTestProvider = FutureProvider.autoDispose
        .family<TabsState, BuildContext>((ref, context) async {
      final tabs = await ref.watch(tabsProvider(widget.spaceId).future);
      **final selectedIndex = ref.watch(selectedTabIdxProvider(widget.spaceId));**
      final ctrl = TabController(
        length: tabs.length,
        vsync: this,
        **initialIndex: selectedIndex.valueOrNull ?? 0,**
      );
      ref.onDispose(() {
        ctrl.dispose(); // we need to clean this up.
      });
      return TabsState(tabs, ctrl);
    });
  }
```

Reply:
okay. and that works as wanted for the scenario I described? Click on "see all spaces" on the overview page gets you to the subpage without flickering, too? If you tested that and confirm it, all good!

Reply:
Are you talking about "See all spaces" button on home overview page?
Let me share reference video so that you can check this work.

Uploading Screen Recording 2023-11-15 at 6.31.27 PM.mov…

Reply:
Let me know if above video is not covering the scenario you are describing.

Reply:
looks good.